src/mpsc/blocking.rs

```diff
@@ -16,6 +16,8 @@ use crate::{
 };
 use core::{fmt, pin::Pin};
 use errors::*;
+use std::sync::mpsc::RecvTimeoutError;
```

Comment:
Since this function is only being added for the blocking module, I've reused `std::sync::mpsc`'s error, but if required, I can create a separate error type for this crate.